File: algorithms/cracking_the_coding_interview/sorting_bubble_sort.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	reader.ReadString('\n')

	line, _ := reader.ReadString('\n')
	line = strings.TrimSpace(line)
	arrayOfStrings := strings.Fields(line)
	length := len(arrayOfStrings)

	array := make([]int, length)
	for i := range arrayOfStrings {
		array[i], _ = strconv.Atoi(arrayOfStrings[i])
	}

	swaps := 0
	for i := 0; i < length; i++ {
		for j := 0; j < length-1; j++ {
			if array[j] > array[j+1] {
				swaps++

				temp := array[j]
				array[j] = array[j+1]
				array[j+1] = temp
			}
		}
	}

	fmt.Printf("Array is sorted in %d swaps. \n", swaps)
	fmt.Printf("First Element: %d \n", array[0])
	fmt.Printf("Last Element: %d \n", array[length-1])
}

File: algorithms/cracking_the_coding_interview/sorting_bubble_sort.php
<?php
declare(strict_types=1);

$handle = fopen('php://stdin', 'r');
fscanf($handle, '%d',$n);
$array = explode(' ', fgets($handle));
array_walk($array, 'intval');

$sortedArray = bubbleSort($array);
echo "First Element: " . reset($sortedArray) . " \n";
echo "Last Element: " . end($sortedArray) . " \n";

/**
 * @param array $array
 * @return array
 */
function bubbleSort(array $array): array
{
    $swaps = 0;
    $length = count($array);

    for ($i = 0; $i < $length; $i++) {
        for ($j = 0; $j < $length - 1; $j++) {
            // Swap adjacent elements if they are in decreasing order
            if ($array[$j] > $array[$j + 1]) {
                $swaps++;

                $temp = $array[$j];
                $array[$j] = $array[$j + 1];
                $array[$j + 1] = $temp;
            }
        }
    }

    echo "Array is sorted in $swaps swaps. \n";

    return $array;
}